src/EditorFeatures/CSharp/NamedArguments/NamedArgumentsAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Linq;
+
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.NamedArguments
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class NamedArgumentsAnalyzer : DiagnosticAnalyzer
```

Comment:
Do we need an analyzer for this?  It seems like just a CodeRefactoringProvider would work for this.